ARCs/arc-0056.md

```diff
@@ -0,0 +1,381 @@
+   */
+  networks?: {
+    /**
+     * The key is the base64 genesis hash of the network, and the value contains
+     * information about the deployed contract in the network indicated by the
+     * key. A key containing the human-readable name of the network MAY be
+     * included, but the corresponding genesis hash key MUST also be defined
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
+    };
+  };
+  /** Named structs use by the application */
+  structs: { [structName: StructName]: StructFields };
+  /** All of the methods that the contract implements */
+  methods: Method[];
+  state: {
+    /** Defines the values that should be used for GlobalNumUint, GlobalNumByteSlice, LocalNumUint, and LocalNumByteSlice when creating the application  */
+    schema: {
+      global: {
+        ints: number;
+        bytes: number;
+      };
+      local: {
+        ints: number;
+        bytes: number;
+      };
+    };
+    /** Mapping of human-readable names to StorageKey objects */
+    keys: {
+      global: { [name: string]: StorageKey };
+      local: { [name: string]: StorageKey };
+      box: { [name: string]: StorageKey };
+    };
+    /** Mapping of human-readable names to StorageMap objects */
+    maps: {
+      global: { [name: string]: StorageMap };
+      local: { [name: string]: StorageMap };
+      box: { [name: string]: StorageMap };
+    };
+  };
+  /** Supported bare actions for the contract. An action is a combination of call/create and an OnComplete */
+  bareActions: {
+    /** OnCompletes this method allows when appID === 0 */
+    create: ("NoOp" | "OptIn" | "DeleteApplication")[];
+    /** OnCompletes this method allows when appID !== 0 */
+    call: (
+      | "NoOp"
+      | "OptIn"
+      | "CloseOut"
+      | "ClearState"
+      | "UpdateApplication"
+      | "DeleteApplication"
+    )[];
+  };
+  /** Information about the TEAL programs */
+  sourceInfo?: {
+    /** Approval program information */
+    approval: SourceInfo[];
+    /** Clear program information */
+    clear: SourceInfo[];
+  };
+  /** The pre-compiled TEAL that may contain template variables. MUST be omitted if included as part of ARC23 */
+  source?: {
+    /** The approval program */
+    approval: string;
+    /** The clear program */
+    clear: string;
+  };
+  /** The compiled bytecode for the application. MUST be omitted if included as part of ARC23 */
+  byteCode?: {
+    /** The approval program */
+    approval: string;
+    /** The clear program */
+    clear: string;
+  };
+  /** Information used to get the given byteCode and/or PC values in sourceInfo. MUST be given if byteCode or PC values are present */
+  compilerInfo?: {
+    /** The name of the compiler */
+    compiler: "algod" | "puya";
+    /** Compiler version information */
+    compilerVersion: {
+      major: number;
+      minor: number;
+      patch: number;
+      commit?: string;
```

Comment:
tealscript emits `commitHash` rather than `commit`, which I think is a better property name. Should we change?

Reply:
Yeah I will update the ARC to use `commitHash`